plugins/connection/aws_ssm.py

```diff
@@ -534,7 +534,13 @@ def _flush_stderr(self, subprocess):
 
     def _get_url(self, client_method, bucket_name, out_path, http_method, profile_name, extra_args=None):
         ''' Generate URL for get_object / put_object '''
-        region_name = self.get_option('region') or 'us-east-1'
+
+        client_s3 = boto3.client('s3')
+        bucket_location = client_s3.get_bucket_location(
+            Bucket=(self.get_option('bucket_name')),
+        )
+        region_name = bucket_location['LocationConstraint']
```

Comment:
Looks good - but I believe the addressing style needs to be hardcoded to virtual to work with regioned buckets. See https://docs.aws.amazon.com/cli/latest/topic/s3-config.html#addressing-style and https://github.com/ansible-collections/community.aws/pull/743/files

Reply:
Thanks @lblakey - Corporate IP policies means I cannot merge PRs unfort.